sdk/identity/identity/Troubleshooting.md

```diff
@@ -0,0 +1,241 @@
+| AZURE PUBLIC CLOUD | https://login.microsoftonline.com/ |
+| AZURE GERMANY      | https://login.microsoftonline.de/  |
+| AZURE CHINA        | https://login.chinacloudapi.cn/    |
+| AZURE GOVERNMENT   | https://login.microsoftonline.us/  |
+
+## Troubleshooting Azure CLI Authentication Issues
+
+### Credential Unavailable
+
+#### Azure CLI Not Installed.
+
+THe `Azure CLI Credential` failed to execute as Azure CLI command line tool is not installed.
+To use Azure CLI credential, the Azure CLI needs to be installed, please follow the instructions [here](https://docs.microsoft.com/cli/azure/install-azure-cli)
+to install it for your platform and then try running the credential again.
+
+#### Azure account not logged in.
+
+The `Azure CLI Credential` utilizes the current logged in Azure user in Azure CLI to fetch an access token.
+You need to login to your account in Azure CLI via `az login` command. You can further read instructions to [Sign in with Azure CLI](https://docs.microsoft.com/cli/azure/authenticate-azure-cli).
+Once logged in try running the credential again.
+
+## Troubleshooting Azure Powershell Authentication Issues
+
+### Credential Unavailable
+
+#### Powershell not installed.
+
+The `Azure Powershell Credential` utilizes the locally installed `Powershell` command line tool to fetch an access token. Please ensure it is installed on your platform by following the instructions [here](https://docs.microsoft.com/powershell/scripting/install/installing-powershell?view=powershell-7.1) and then run the credential again.
+
+#### Azure Az Moudle Not Installed.
+
+The `Azure Powershell Credential` failed to execute as Azure az module is not installed.
+To use Azure Powershell credential, the Azure az module needs to be installed, please follow the instructions [here](https://docs.microsoft.com/powershell/azure/install-az-ps?view=azps-6.3.0)
+to install it for your platform and then try running the credential again.
+
+#### Azure account not logged in.
+
+The `Azure Powershell Credential` utilizes the current logged in Azure user in Azure Powershell to fetch an access token.
+You need to login to your account in Azure Powershell via `Connect-AzAccount` command. You can further read instructions to [Sign in with Azure Powershell](https://docs.microsoft.com/powershell/azure/authenticate-azureps?view=azps-6.3.0).
+Once logged in try running the credential again.
+
+#### Deserialization error.
+
+The `Azure Powershell Credential` was able to retrieve a response from the Azure Powershell when attempting to get an access token but failed
+to parse that response.
+In your local powershell window, run the following command to ensure that Azure Powerhsell is returning an access token in correct format.
+
+```pwsh
+Get-AzAccessToken -ResourceUrl "<Scopes-Url>"
+```
+
+In the event above command is not working properly, follow the instructions to resolve the Azure Powershell issue being faced and then try running the credential again.
```

Comment:
I believe we should close asking users to log issues and perhaps linking them to our contribution guide.